src/components/clusterConfiguration/ClusterDefaultConfigurationContext.tsx

```diff
@@ -1,31 +1,52 @@
 import _ from 'lodash';
-import React, { useContext, useEffect, useState } from 'react';
-import { handleApiError, getErrorMessage } from '../../api';
+import React, { PropsWithChildren, useContext, useEffect } from 'react';
 import { getClustersDefaultConfiguration } from '../../api/clusters';
 import { ClusterDefaultConfig } from '../../api/types';
-import { AlertsContext } from '../AlertsContextProvider';
+
+type State<T> = { status: 'idle' | 'loading' | 'succeeded' | 'failed'; data: T };
+type Actions<T> = { type: 'request' } | { type: 'failure' | 'success'; data: T };
+
+const reducer: React.Reducer<State<ClusterDefaultConfig>, Actions<ClusterDefaultConfig>> = (
+  state,
+  action,
+) => {
+  let newState: State<ClusterDefaultConfig>;
+  switch (action.type) {
+    case 'request':
+      newState = { status: 'loading', data: {} };
+      break;
+    case 'success':
+      newState = { status: 'succeeded', data: action.data };
+      break;
+    case 'failure':
+      newState = { status: 'failed', data: action.data };
+      break;
+    default:
+      newState = state;
+      break;
+  }
+
+  return newState;
```

Comment:
This could be simplified to:

```suggestion
  switch (action.type) {
    case 'request':
      return { status: 'loading', data: {} };
    case 'success':
      return { status: 'succeeded', data: action.data };
    case 'failure':
      return { status: 'failed', data: action.data };
    default:
      return state;
  }
```

Reply:
Agree. IMHO I don't like many `return` statements. I see this as an optimization which is performed anyway by webpack at build time.